web_poet/_typing.py

```diff
@@ -18,7 +18,16 @@ def is_generic_alias(obj) -> bool:
 
 
 def get_generic_parameter(cls):
-    for base in cls.__orig_bases__:
+    for base in getattr(cls, "__orig_bases__", []):
         if is_generic_alias(base):
             args = _get_args(base)
             return args[0]
+
+
+def get_item_cls(cls, default=None, preferred=None):
+    if preferred:
+        return preferred
```

Comment:
I'd remove the `preferred` argument, and handle it on a caller side, but that's very minor; the current code is fine as well.

Reply:
Updated in https://github.com/scrapinghub/web-poet/pull/84/commits/928188e2c1267654ff12492e129cc6ecf0a7e251